### ISSUE TITLE
False-positive inconsistent overload implementation with `Callable` and `TypeVar`

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pyright#2792]
**Describe the bug**
```
error: Overloaded function implementation is not consistent with signature of overload 1
  Function return type "Future[_R@run_job] | None" is incompatible with type "Future[_R@run_job | Awaitable[_R@run_job]] | None"
    Type "Future[_R@run_job] | None" cannot be assigned to type "Future[_R@run_job | Awaitable[_R@run_job]] | None"
      Type "Future[_R@run_job]" cannot be assigned to type "Future[_R@run_job | Awaitable[_R@run_job]] | None"
        TypeVar "_T@Future" is invariant
          Type "_R@run_job" cannot be assigned to type "_R@run_job | Awaitable[_R@run_job]"
        Type cannot be assigned to type "None" (reportGeneralTypeIssues)
```

**To Reproduce**
_See code below_

**Expected behavior**
_No error, same as with mypy 0.930._

**Screenshots or Code**
```python
import asyncio
from typing import Any, Awaitable, Callable, TypeVar, overload

_R = TypeVar("_R")

@overload
def run_job(
    target: Callable[..., Awaitable[_R]], *args: Any
) -> asyncio.Future[_R] | None:
    ...

@overload
def run_job(
    target: Callable[..., _R], *args: Any
) -> asyncio.Future[_R] | None:
    ...

def run_job(
    target: Callable[..., Awaitable[_R]] | Callable[..., _R], *args: Any
) -> asyncio.Future[_R] | None:
    ...
```

**VS Code extension or command-line**
Command-line, v1.1.202

**Additional context**
Technically speaking, the overloads aren't even needed. The return type can be inferred by pyright even without them. Unfortunately, there is a bug in mypy though. To get around it, the overloads are required.

```py
def func() -> int:
    return 42

async def async_func_int() -> int:
    return 42

async def async_func_none() -> None:
    return None

reveal_type(run_job(func))
reveal_type(run_job(async_func_int))
reveal_type(run_job(async_func_none))
```

<details>
<summary>Mypy - without overloads</summary>

```
note: Revealed type is "Union[asyncio.futures.Future[builtins.int*], None]"
note: Revealed type is "Union[asyncio.futures.Future[<nothing>], None]"
error: Argument 1 to "run_job" has incompatible type "Callable[[], Coroutine[Any, Any, int]]"; expected "Callable[..., Awaitable[<nothing>]]"
note: Revealed type is "Union[asyncio.futures.Future[<nothing>], None]"
error: Argument 1 to "run_job" has incompatible type "Callable[[], Coroutine[Any, Any, None]]"; expected "Callable[..., Awaitable[<nothing>]]"
```
</details>

<details>
<summary>Mypy - with overloads</summary>

```
note: Revealed type is "Union[asyncio.futures.Future[builtins.int*], None]"
note: Revealed type is "Union[asyncio.futures.Future[builtins.int*], None]"
note: Revealed type is "Union[asyncio.futures.Future[None], None]"
```
</details>

<details>
<summary>Pyright - without overloads</summary>

```
info: Type of "run_job(func)" is "Future[int] | None"
info: Type of "run_job(async_func_int)" is "Future[int] | None"
info: Type of "run_job(async_func_none)" is "Future[None] | None"
```
</details>

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.206, which I just published. It will also be in the next release of pylance.
